packages/logging-interceptor/src/logging.interceptor.ts

```diff
@@ -173,31 +173,34 @@ export class LoggingInterceptor implements NestInterceptor {
    * @returns the masked data
    */
   private maskData(data: unknown, maskingOptions: string[] | true, path: string = ''): unknown {
+    // Parse the data to avoid having constructors like new ObjectId() in the body
+    const parsedData = JSON.parse(JSON.stringify(data));
```

Comment:
I think, it should handle circular references in the stringify.
```suggestion
    const parsedData = JSON.parse(JSON.stringify(data, handleCircularReferences));
```

Reply:
Is this an option in stringify ?

Reply:
You can pass a `replacer` in `JSON.stringify` to detect circular references and replace them by reference ID to prevent from having an infinite loop during the stringify.

- [replacer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#replacer)
- [example of handler](https://stackoverflow.com/a/11616993/6086598)

Reply:
I used a package that deals with the handling of circular references